oshmem/mca/atomic/ucx/atomic_ucx_cswap.c

```diff
@@ -27,44 +27,48 @@ int mca_atomic_ucx_cswap(void *target,
                          int pe)
 {
     ucs_status_t status;
+    ucs_status_ptr_t status_ptr;
     spml_ucx_mkey_t *ucx_mkey;
     uint64_t rva;
+    uint64_t val;
+    uint64_t cmp;
+
+    if (8 == nlong) {
+        val = *(uint64_t*)value;
+    } else if (4 == nlong) {
+        val = *(uint32_t*)value;
+    } else {
+        ATOMIC_ERROR("[#%d] Type size must be 4 or 8 bytes.", my_pe);
+        return OSHMEM_ERROR;
+    }
 
     ucx_mkey = mca_spml_ucx_get_mkey(pe, target, (void *)&rva); 
     if (NULL == cond) {
-        switch (nlong) {
-            case 4:
-                status = ucp_atomic_swap32(mca_spml_self->ucp_peers[pe].ucp_conn, 
-                        *(uint32_t *)value, rva, ucx_mkey->rkey, prev);
-                break;
-            case 8:
-                status = ucp_atomic_swap64(mca_spml_self->ucp_peers[pe].ucp_conn, 
-                        *(uint64_t *)value, rva, ucx_mkey->rkey, prev);
-                break;
-            default:
-                goto err_size;
-        }
+        status_ptr = ucp_atomic_fetch_nb(mca_spml_self->ucp_peers[pe].ucp_conn,
+                                         UCP_ATOMIC_FETCH_OP_SWAP, val, prev, nlong,
+                                         rva, ucx_mkey->rkey, mca_atomic_ucx_complete_cb);
+        status = mca_atomic_ucx_wait_request(status_ptr);
     }
     else {
-        switch (nlong) {
-            case 4:
-                status = ucp_atomic_cswap32(mca_spml_self->ucp_peers[pe].ucp_conn, 
-                        *(uint32_t *)cond, *(uint32_t *)value, rva, ucx_mkey->rkey, prev);
-                break;
-            case 8:
-                status = ucp_atomic_cswap64(mca_spml_self->ucp_peers[pe].ucp_conn, 
-                        *(uint64_t *)cond, *(uint64_t *)value, rva, ucx_mkey->rkey, prev);
-                break;
-            default:
-                goto err_size;
+        if (8 == nlong) {
```

Comment:
let's try to branch on "nlong" only once. currently, for cswap, it's done 3 times

Reply:
ok, only one used